boolean/boolean.py

```diff
@@ -866,7 +866,12 @@ def __lt__(self, other):
     def __gt__(self, other):
         lt = other.__lt__(self)
         if lt is NotImplemented:
-            return not self.__lt__(other)
+            self_lt = self.__lt__(other)
+            if type(self_lt) is type(NotImplemented):
```

Comment:
What is the reason to check the type?

```suggestion
            if self_lt is NotImplemented:
```